### ISSUE TITLE
Feature: Reintroduce FogBugz

### ISSUE THREAD

**Issue by MadsNielsen** [Praqma/Praqmatic-Automated-Changelog#62]

**Comment by buep** [comment]
You need to write tests! And the same comments about issue description and workflows apply as my comment in: 

https://github.com/Praqma/Praqmatic-Automated-Changelog/issues/63#issuecomment-236559443

**Comment by MadsNielsen** [comment]
Most of the requirements for this are already in place. I've got a simple test, that test the configuration part of fogbugz section. For me, what is missing in this case is a test with some mocked return data from FogBugz. 

That is a small unit test, fogbugz works the same as jira.

**Comment by buep** [comment]
Which of the above solutions you suggested did you chose?

Which test scenarios do you have with Fogbugz?

**Comment by MadsNielsen** [comment]
1. I used the 'no-auth' easy solution with warning in console, fogbugz does not support basic auth (token security as part of the query string)
2. Theres a very simple XML library i used to convert output to ruby hash. (Oneliner... `XmlSimple.xml_in...` 

The only scenario i have is that the configuration is parsed correctly and the task system is picked up and fails  correctly. Otherwise it's identical to Jira. It's the same as with GitHub which makes it very simple to integrate, since they're all using REST api's and are very similar. My test scenario for fogbugz would likely just include a sample return snippet of fogbugz XML, to make sure that my model objects are created correctly. We could add a unit test that creates a report with that model object if that is required.

**Comment by buep** [comment]
I'm missing test scenarios around Fogbugz related to communication... if not responding, if timing out, if bad data received .... all these situation needs to be tested, either for fogbugz or in general and report decent error messages from PAC.

If Fogbugz have a version of their api, and you can actually see that, I'm okay with you stub the data. Then just make sure if the version for fogbugz data isn't as expected, we fail properly. That way we know it works in the future, or a least report decent error about the expected dataformat doesn't match and FB might have upgrade their api.

**Comment by buep** [comment]
yep, good scenario. I want them. Make the general, so they can used for all the tasks systems. I guess only the 200 Ok and third with broken data is task specific.
